swap/swap.go

```diff
@@ -145,8 +174,8 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 
 	// Check if balance with peer is over the disconnect threshold
 	balance := swapPeer.getBalance()
-	if balance >= s.disconnectThreshold {
-		return fmt.Errorf("balance for peer %s is over the disconnect threshold %d, disconnecting", peer.ID().String(), s.disconnectThreshold)
+	if disconnectThreshold := int64(s.disconnectThreshold); balance >= disconnectThreshold {
```

Comment:
?

`s.disconnectThreshold` is already `int64`, why this conversion and this new assignment? also, don't use a different variable here, this runs A LOT, so even if it is in the nanoseconds range, I'd avoid all this conversion and new variable stuff, just leave it as it was

Reply:
the thresholds are originally of the `uint` type, so a comparison between any of them and `balance`—which is a signed integer—will not compile

Reply:
There are some options here:

- Make the thresholds an `int64`, just as `balance`. 
Advantage: direct comparison is possible, disadvantage: a non-positive threshold does not make a lot of sense. Possible edge-cases which must be explicitly taken into account.
- Get rid of variable assignments. Advantage: efficiency. Disadvantage: code-readability
- Leave it as is. Advantage: code-readability. Disadvantage: efficiency-loss due to variable assignments/conversion.

I would leave it as-is and mark it as a TODO for the future that we can possibly have more efficiency here.

Reply:
I disagree, you are introducing a change here so it should not be left to the future.

The threshold is a constant, there is no point in making the conversion all the time. I would suggest to change `s.disconnectThreshold` a `int64` as it was, and if you want it to be a `uint64` in the flag, which is justified because you want to avoid the user having to think if it's negative or positive, then make the conversion only once when it is set in the `Swap` object.

Reply:
Done, good suggestion!